### ISSUE TITLE
If steam message content contains \n will lost content

### ISSUE THREAD

**Issue by tomtrije** [mpetazzoni/sse.js#33]
likes title describle

**Comment by tomtrije** [comment]
Bug is here:

`chunk.split(/\n|\r\n|\r/).forEach(function(line) {
      line = line.trimRight();
      var index = line.indexOf(this.FIELD_SEPARATOR);
      if (index <= 0) {
        // Line was either empty, or started with a separator and is a comment.
        // Either way, ignore.
        return;
      }

      var field = line.substring(0, index);
      if (!(field in e)) {
        return;
      }

      var value = line.substring(index + 1).trimLeft();
      if (field === 'data') {
        e[field] += value;
      } else {
        e[field] = value;
      }
    }.bind(this));`

**Comment by tomtrije** [comment]
if chunk data likes this, will lost \n(which in line3 and line4) in "data" field
```
event:add
id:7753781597103745505
data:
data:
data:something
```

**Comment by tomtrije** [comment]
also `trimRight` and `trimLeft` will cause  lost blank character

**Comment by visceraldesert** [comment]
> also `trimRight` and `trimLeft` will cause lost blank character

I just hit that problem in my implementation, I lost important blank spaces

**Comment by rodrigovr** [comment]
@tomtrije You can try my changes from PR https://github.com/mpetazzoni/sse.js/pull/35

my repo is at https://github.com/rodrigovr/sse.js/tree/main but I pretend to close the fork as soon as the changes are accepted here

**Comment by mpetazzoni** [comment]
Merged #35. Thanks @rodrigovr !
